### DISCUSSION TITLE
To ignore the health check of a specific Deployment

### DISCUSSION THREAD

**Discussion by stoneeve415** [argoproj/argo-cd#19257] [General]
Hello everyone, I need help. I want to ignore the status of a specific deployment (which is expected to be in progress).
**Attempt 1**:
```
Add argocd.argoproj.io/compare-options: IgnoreExtraneous to the deployment, as shown below:
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-app
  annotations:
    argocd.argoproj.io/compare-options: IgnoreExtraneous
spec:
  ...
```
But this does not affect the health status.

**Attempt 2**:
Customize health checks using resource.customizations, and add the skip-health label to the deployment that needs to skip health checks.
```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: argocd
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
  name: argocd-cm
data:
  resource.customizations: |
    apps/Deployment:
      health.lua: |
        local health_check = {}
        health_check.status = "Progressing"
        health_check.message = "replication controller is waiting for pods to run"
        if obj.metadata.labels ~= nil and type(obj.metadata.labels) == "table" and obj.metadata.labels["skip-health"] == "true" then
          health_check.status = "Healthy"
          health_check.message = "Application is progressing"
          return health_check
        return nil
```
However, returning nil will cause all other deployments to be in the progressing state. Is there a way to directly invoke the default health check logic?

**Comment by ChristianCiach** [comment]
> Is there a way to directly invoke the default health check logic?

Looking at the code, the answer is a clear "no". But I like the idea and it seems easy enough to implement.

Currently, returning `nil` is handled the same as returning `{}`. Maybe this should be changed? Maybe returning `nil` should fall back to the default health check, but this *may* be a breaking change. To avoid such a breaking change, maybe we should make it more explicit:

```lua
result = {}
result.status = "Default"
return result
```

... or something like that.

**Reply by stoneeve415** [reply]
![企业微信截图_17221700445307](https://github.com/user-attachments/assets/7a3fc20c-d193-4d4e-b4ac-37f0beed4e30)
First of all, thank you for your appreciation. However, the official documentation indicates that only four states are supported, and `returning default` would cause other app statuses to become unknown. I tried implementing the health check logic for the deployment using `Lua`, but it seems a bit complex and may lead to issues. Is there a way to directly invoke the default health check logic of Argo CD?
